app/common/src/utilities/data/object.ts

```diff
@@ -15,11 +15,15 @@ export function unsafeKeyValuePair<Key extends PropertyKey, Value>(key: Key, val
 
 /**
  * Immutably shallowly merge an object with a partial update.
- * Does not preserve classes. Useful for preserving order of properties.
+ *
+ * Does not preserve classes. Useful for preserving order of properties. Output may alias input.
  */
 export function merge<T extends object>(object: T, update: Partial<T>): T {
-  for (const [key, value] of Object.entries(update)) {
-    if (!Object.is(value, (object as Record<string, unknown>)[key])) {
+  for (const key of Reflect.ownKeys(update)) {
+    if (
+      !(key in object) ||
+      !Object.is(update[key as keyof typeof update], object[key as keyof typeof object])
```

Comment:
`typeof object` is just `T`, isn't it?
```suggestion
      !Object.is(update[key as keyof T], object[key as keyof T])
```
Didn't check that though, it's possible typescript doesn't like it.